parlai/tasks/wikipedia/agents.py

```diff
@@ -76,6 +78,73 @@ def get_extraction_instructions(self):
         return instructions
 
 
+class FullSplitTeacher(ChunkTeacher):
+    """
+    Full Wikipedia teacher that splits the chunks into train/valid/test.
+    """
+
+    def _get_data_folder(self):
+        return os.path.join(self.opt['datapath'], 'wikipedia/full/wiki_full_extracted')
+
+    def _get_num_samples(self, datatype) -> int:
+        """
+        Return the number of samples given the datatype.
+        """
+        if 'train' in datatype:
+            return 5437097
```

Comment:
Hmm, might be a good to have a unit test for this teacher as well (if it doesn't already exist) to check that this number of samples remains accurate